eglot-x.el

```diff
@@ -2036,5 +2052,46 @@ Use `browse-url' for non-local schemas."
       (find-file-noselect related-file) ; See Emacs bug#57325.
       (list related-file))))
 
+;; inactive code suport
+(cl-defmethod eglot-handle-notification
+  (_server (_method (eql $ccls/publishSkippedRanges)) &key uri skippedRanges)
+  "Listen to CCLS's skipped region notifications and deemphasize
+ inactive code"
+  (when-let* (eglot-x-enable-inactive-code
+	      (buffer (or (find-buffer-visiting (eglot--uri-to-path uri))
+			  (gethash uri eglot--temp-location-buffers))))
+    (with-current-buffer
+	buffer
+      (remove-overlays nil nil 'inactive--code t)
+      (mapc (lambda (range)
+	      (pcase-let*
+		  ((`(,beg . ,end) (eglot--range-region range)))
+		(let ((ov (make-overlay beg end buffer t nil)))
+		  (overlay-put ov 'face 'font-lock-comment-face)
+		  (overlay-put ov 'eglot--overlay t)
+		  (overlay-put ov 'inactive--code t))))
+	    skippedRanges))))
+
+(cl-defmethod eglot-handle-notification
+  (_server (_method (eql textDocument/inactiveRegions))
+           &key regions textDocument &allow-other-keys)
+  "Listen to clangd's inactive code notifications and deemphasize
+ inactive code (adapted from example code in eglot's user
```

Comment:
I think it is better to put the "adapted from ..." part into a simple comment and remove from the docstring.  (Also first line should be a full sentence.)